app/src/main/java/com/example/tasky/agenda_details/presentation/AgendaDetailsViewModel.kt

```diff
@@ -141,23 +132,22 @@ class AgendaDetailsViewModel @Inject constructor(
     }
 
     fun onAddPhotosClick(photoUris: List<Uri>) {
-        _photosUri.value = photoUris
+        compressAndAddImage(photoUris)
     }
 
-    fun compressAndAddImage(context: Context, uris: List<Uri>) {
+    private fun compressAndAddImage(uris: List<Uri>) {
         viewModelScope.launch {
             var skipped = 0
             val newCompressedList = uris.mapNotNull { uri ->
-                val drawable = imageCompressionUseCase.uriToDrawable(context, uri)
+                val drawable = imageCompressionUseCase.uriToDrawable(uri)
                 val originalBitmap = (drawable as BitmapDrawable).bitmap
                 Log.d(TAG, "Original Bitmap Size in bytes: ${originalBitmap.byteCount}")
 
-                val compressedByteArray = withContext(Dispatchers.IO) {
-                    imageCompressionUseCase.compressImage(drawable)
-                }
+                val compressedByteArray = imageCompressionUseCase.compressImage(drawable)
 
-                val compressedBitmap =
+                val compressedBitmap = withContext(Dispatchers.IO) {
                     BitmapFactory.decodeByteArray(compressedByteArray, 0, compressedByteArray.size)
+                }
```

Comment:
It's not just the Context that should stay out of ViewModels, but any class from the Android SDK (like Bitmaps), since these classes can't be used in local unit tests. 

I don't think you need a bitmap here at all, since you could just pass the compressed byte array to the UI which loads it as an image (I think Coil does that out of the box for byte arrays)